azurerm/internal/services/maintenance/maintenance_configuration_resource.go

```diff
@@ -62,26 +63,75 @@ func resourceArmMaintenanceConfiguration() *pluginsdk.Resource {
 			"scope": {
 				Type:     pluginsdk.TypeString,
 				Optional: true,
-				Default:  string(maintenance.ScopeAll),
+				Default:  "All",
 				ValidateFunc: validation.StringInSlice([]string{
-					string(maintenance.ScopeAll),
+					"All", // All is still accepted by the API
+					string(maintenance.ScopeExtension),
 					string(maintenance.ScopeHost),
-					string(maintenance.ScopeInResource),
-					string(maintenance.ScopeResource),
+					string(maintenance.ScopeInGuestPatch),
+					string(maintenance.ScopeOSImage),
+					string(maintenance.ScopeSQLDB),
+					string(maintenance.ScopeSQLManagedInstance),
 				}, false),
 			},
 
-			// There's a bug in the Azure API where the the key of tags is returned in lower-case
-			// BUG: https://github.com/Azure/azure-rest-api-specs/issues/9075
-			// use custom tags defition here to prevent inputting upper case key
-			"tags": {
-				Type:         pluginsdk.TypeMap,
-				Optional:     true,
-				ValidateFunc: validate.TagsWithLowerCaseKey,
+			"visibility": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Default:  string(maintenance.VisibilityCustom),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(maintenance.VisibilityCustom),
+					// Creating public configurations doesn't appear to be supported, API returns `Public Maintenance Configuration must set correct properties`
+					// string(maintenance.VisibilityPublic),
+				}, false),
+			},
+
+			"window": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"start_date_time": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+						},
+						"expiration_date_time": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+						},
+						"duration": {
```

Comment:
what happens if duration is greater then the end time? ca nwe set both?

Reply:
See below - I don't think the duration is related to the expiry date